File: goblinframework-core/src/main/java/org/goblinframework/core/conversion/converter/InstantToLongConverter.java
package org.goblinframework.core.conversion.converter;

import org.goblinframework.api.annotation.Install;
import org.jetbrains.annotations.NotNull;
import org.springframework.core.convert.converter.Converter;

import java.time.Instant;

@Install
final public class InstantToLongConverter implements Converter<Instant, Long> {

  @NotNull
  @Override
  public Long convert(@NotNull Instant source) {
    return source.toEpochMilli();
  }
}

File: goblinframework-core/src/main/java/org/goblinframework/core/conversion/converter/InstantToLongConverter.kt
package org.goblinframework.core.conversion.converter

import org.springframework.core.convert.converter.Converter

import java.time.Instant

class InstantToLongConverter : Converter<Instant, Long> {

  override fun convert(source: Instant): Long? {
    return source.toEpochMilli()
  }
}
